### PR TITLE
feat: introduce enableAppbase prop to enable RS-API

### PR DESCRIPTION
**Pull request by ShahAnuj2610**
how to test:

run following commands in sequence:
```
git checkout feat/rs-api
```

```
yarn && yarn dev:react
```

go to `packages/react-searchbox/examples/rs-api` and run
```
yarn start
```

### CODE REVIEW

File: packages/react-searchbox/examples/basic/src/App.js [hunk 1]
```diff
@@ -4,27 +4,37 @@ import SearchBox from '@appbaseio/react-searchbox';
 import { FaMicrophone } from 'react-icons/fa';
 
 export default class App extends Component {
+  state = { value: '' };
```
Comment (bietkul) [inline packages/react-searchbox/examples/basic/src/App.js]: @ShahAnuj2610 Why did you edit the old examples, Can you please add new examples with RS API?

File: packages/searchbase/src/index.js [hunk 2]
```diff
@@ -273,17 +278,17 @@ class Searchbase {
     this.dataField = dataField;
     this.aggregationField = aggregationField;
     this.credentials = credentials || '';
-    this.nestedField = nestedField || '';
+    this.nestedField = nestedField;
```
Comment (bietkul) [inline packages/searchbase/src/index.js:281]: @ShahAnuj2610 Are you using `flow`, why do you need to change it?

File: packages/searchbase/src/index.js [hunk 3]
```diff
@@ -857,6 +871,40 @@ class Searchbase {
...
+      'query',
+      prevQuery,
+      this._query
+    );
+  }
+
+  getAppbaseResultQuery() {
+    return {
+      id: 'SearchResult',
+      dataField: this.dataField,
```
Comment (bietkul) [inline packages/searchbase/src/index.js]: @ShahAnuj2610 Validate `dataField`, it must be an `array`.

File: packages/searchbase/src/index.js [hunk 4]
```diff
@@ -893,17 +941,46 @@ class Searchbase {
...
-      this._query
+      this._suggestionsQuery
     );
   }
 
-  _updateSuggestionsQuery(query?: Object, queryOptions?: Object): void {
+  getAppbaseSuggestionsQuery() {
+    return {
+      id: 'DataSearch',
+      dataField: this.dataField,
```
Comment (bietkul) [inline packages/searchbase/src/index.js]: @ShahAnuj2610 Validate

File: packages/searchbase/src/index.js [hunk 5]
```diff
@@ -893,17 +941,46 @@ class Searchbase {
...
+      from: this.from,
+      size: this.size,
+      sortBy: this.sortBy,
+      aggregationField: this.aggregationField,
+      includeFields: this.includeFields,
+      excludeFields: this.excludeFields,
+      fuzziness: this.fuzziness,
+      searchOperators: this.searchOperators,
+      highlight: this.highlight,
+      highlightField: this.highlightField
```
Comment (bietkul) [inline packages/searchbase/src/index.js:1012]: @ShahAnuj2610 Do we not support `customHighlight` in `searchbase`?

**Review by bietkul — CHANGES_REQUESTED** [review]
1. No changes in `searchbox(JS)`?
2. `enableQueryRules` is missing

**Comment by ShahAnuj2610** [comment]
@bietkul for `searchbox` we just need to update `@appbaseio/searchbase` version

File: packages/searchbase/src/index.js (hunk 5 above)
Comment (ShahAnuj2610) [reply packages/searchbase/src/index.js:1012]: no, we don't

**Comment by bietkul** [comment]
@ShahAnuj2610 `settings` is also missing in the request body

File: packages/searchbase/src/index.js (hunk 2 above)
Comment (ShahAnuj2610) [reply packages/searchbase/src/index.js:281]: keeping it as an empty string affects the `rs query`

**Review by bietkul — APPROVED** [review]

**Pull request merged**